media/js/base/mozilla-traffic-cop.js

```diff
@@ -146,7 +146,21 @@ Mozilla.TrafficCop.prototype.generateRedirectUrl = function(url, querystring) {
     // conjure a random number between 1 and 100 (inclusive)
     var rando = Math.floor(Math.random() * 100) + 1;
 
-    url = url || window.location;
+    // make sure to disregard hash in original URL
+    if (!url) {
+        // build URL root, e.g. https://www.mozilla.org
+        url = window.location.protocol + '//' + window.location.hostname;
+
+        // handy for local testing
+        if (window.location.port !== '') {
```

Comment:
How sure are we that this won't introduce some unexpected results in some browsers? (I have a funny recollection of IE doing some strange things even when no port is specified). Could a simpler option be to use a regex to strip the hash? Either way, an accompanying test would be good here 👍